### PR TITLE
fix(asset-index): use transfer instead of deposit for liquid assets

### LINKED ISSUE #411: Require liquid assets were sent to via XCM
## Issue summary

- Currently `add_asset` mints PINT + liquid assets a like, this assumes they were contributed. Instead, it should be enforced that these are reserve-backed transferred to PINT first.

## Other information and links

-

### PR DESCRIPTION
**Pull request by mattsse**
## Changes

- Fixes a soundness issue mentioned in #331, minting of liquid assets is removed and requires that these assets are deposited via XCM first regardless if it's gov approved.

## Tests

Failing tests are updated to allocate some funds for `add_asset`
Did not check the JS test suite however.

```
cargo t
```

## Issues

- Closes #411

### CODE REVIEW

File: pallets/asset-index/src/lib.rs [hunk 1]
```diff
@@ -1034,8 +1035,8 @@ pub mod pallet {
 			}
 			// native asset can't be added
 			Self::ensure_not_native_asset(&asset_id)?;
-			// mint asset into the treasury account
-			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+			// transfer the asset from the caller to treasury account
+			T::Currency::transfer(asset_id, caller, &Self::treasury_account(), units)?;
```
Comment (clearloop) [inline pallets/asset-index/src/lib.rs:1039]: the caller account may not have enough balance to make transfer

**Review by clearloop — CHANGES_REQUESTED** [review]

File: pallets/asset-index/src/lib.rs (hunk 1 above)
Comment (mattsse) [reply pallets/asset-index/src/lib.rs:1039]: yeah but we catch the error and abort in case that happens?

Comment (clearloop) [reply pallets/asset-index/src/lib.rs:1039]: > The given amount of assets must already exist in the caller's account

After modifying this to `transfer`, we don't have any extrinsic left can mint assets in our pallets, for approving an asset working in PINT, we need to mint the assets with orml pallet by hand, or we'll mint them from the XCM part?

Comment (mattsse) [reply pallets/asset-index/src/lib.rs:1039]: this is intended as discussed here https://github.com/ChainSafe/PINT/pull/401#discussion_r712791918 
Minting liquid assets without receiving them via XCM is a soundness issue

**Comment by clearloop** [comment]
plz fix the benchmarks

**Comment by clearloop** [comment]
Still broken in the benchmarks, I can join to debug this later

**Comment by mattsse** [comment]
> Still broken in the benchmarks, I can join to debug this later

ah I know what's up, fixing this rn

**Review by clearloop — APPROVED** [review]

**Pull request merged**